test/models/user_test.rb

```diff
@@ -78,6 +78,16 @@ def setup
     assert_equal user.lower_login, "user"
   end
 
+  test "user can update username" do
+    user = FactoryGirl.create(:user, :auth_source => auth_sources(:internal))
```

Comment:
No need to add `:auth_source as :internal` as in [user_related.rb](https://github.com/theforeman/foreman/blob/develop/test/factories/user_related.rb) for `:user` factory, it is assigned to `AuthSourceInternal` only.